scene/main/canvas_item.cpp

```diff
@@ -587,7 +587,14 @@ bool CanvasItem::_get(const StringName &p_name, Variant &r_ret) const {
 
 void CanvasItem::_get_property_list(List<PropertyInfo> *p_list) const {
 	List<PropertyInfo> pinfo;
-	RS::get_singleton()->canvas_item_get_instance_shader_parameter_list(get_canvas_item(), &pinfo);
+	if (use_parent_material) {
+		Node *parent = get_parent();
+		CanvasItem *parent_canvas_item = Object::cast_to<CanvasItem>(parent);
+		RS::get_singleton()->canvas_item_get_instance_shader_parameter_list(parent_canvas_item->get_canvas_item(), &pinfo);
```

Comment:
I don't think this is correct. What if the parent has `use_parent_material` enabled as well?

Reply:
In that case they will both use the material of the parent:
<img width="1446" alt="image" src="https://github.com/user-attachments/assets/6dafb4ea-2714-4768-925f-6e84ba123d2d">

<img width="1468" alt="image" src="https://github.com/user-attachments/assets/9526263a-db30-40c2-ada6-47dbfe11f98b">

<img width="1439" alt="image" src="https://github.com/user-attachments/assets/80499fa3-20b3-49b5-8f89-ee5e8979cc98">

Reply:
I am confused now. If I have the following structure
```
Node A
-- Node B
----Node C
```
If Node A has the instance shader applied, Node B recognizes it if `use_parent_material` is enabled and will show the params in the inspector where they can be altered independently from the params of Node A. If Node C now also has `use_parent_material` enabled then it will inherit the material from Node B which itself inherits it from Node A. And the inspector still pops up for both Node B and Node C since they have the `use_parent_material` enabled. If you disable it, the inspector params disappear again.

Reply:
This is how it should be yes, but that isn't how it works right now. Right now both Node B and Node C will inherit the material, but only Node B will be able to modify the parameters in the inspector. 

In this example Icon, Icon2, and Icon3 all have "use_parent_material" enabled. The material is passed down to them, but the uniform set in the inspector is ignored. Then for icon3 the inspector option to set the shader parameter doesn't even appear. 

<img width="1435" alt="Screenshot 2024-11-27 at 12 49 22 PM" src="https://github.com/user-attachments/assets/50fff287-111d-4d9b-833a-44cc0c7619d9">